sdk/ai/ai-projects/src/agents/traceUtility.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TracingSpan } from "@azure/core-tracing";
+import { AgentsApiResponseFormat, AgentsApiResponseFormatOption, MessageContent, ThreadMessage, ThreadMessageOptions } from "./inputOutputs.js";
+import { TracingAttributes, TracingUtility } from "../tracing.js";
+import { getTelemetryOptions } from "../telemetry/telemetry.js";
+
+/**
+ * Adds a message event to the span.
+ * @param span - The span to add the event to.
+ * @param messageAttributes - The attributes of the message event.
+ */
+export function addMessageEvent(span: Omit<TracingSpan, "end">, messageAttributes: ThreadMessageOptions | ThreadMessage,): void {
+
+    const eventBody: Record<string, unknown> = {};
+    const telemetryOptions = getTelemetryOptions()
+    if (telemetryOptions.enableContentRecording) {
+        eventBody.content = getMessageContent(messageAttributes.content);
+    }
+    eventBody.role = messageAttributes.role;
+    if (messageAttributes.attachments) {
+        eventBody.attachments = messageAttributes.attachments.map((attachment) => {
+            return {
+                id: attachment.file_id,
+                tools: attachment.tools.map((tool) => tool.type)
+            };
+        });
+    }
+    const threadId = (messageAttributes as ThreadMessage).thread_id;
+    const agentId = (messageAttributes as ThreadMessage).assistant_id;
+    const threadRunId = (messageAttributes as ThreadMessage).run_id;
+    const messageStatus = (messageAttributes as ThreadMessage).status;
+    const attributes = { eventContent: JSON.stringify(eventBody), threadId, agentId, threadRunId, messageStatus, genAiSystem: TracingAttributes.AZ_AI_AGENT_SYSTEM };
+    TracingUtility.addSpanEvent(span, `gen_ai.${messageAttributes.role}.message`, attributes);
+}
+
+/**
+ * Adds an instruction event to the span.
+ * @param span - The span to add the event to.
+ * @param instructionAttributes - The attributes of the instruction event.
+ */
+export function addInstructionsEvent(span: Omit<TracingSpan, "end">, instructionAttributes: { instructions?: string | null, additional_instructions?: string | null, threadId?: string, agentId?: string }): void {
```

Comment:
Having these helpers is fine, I worry about this `TracingUtility` concept since it duplicates a lot of what is in core-tracing (it may be a nicer API for y'all which is fine if it's all internal)

Reply:
I see now that TracingUtility is an object encapsulating all the state and helpers in static methods. In JS, it is more idiomatic to export the helpers and tracingClient directly from the file. Each file can be treated as a module, and its exports the same as these `static` methods on TracingUtility. This is more of a nit than anything else, so up to you! Maybe you need to wrap these in a TracingUtility for mocking or something...

Reply:
More of a convenience, it's for internal purpose, the only thing it extends is the WithSpan helper providing start and end  and take care of error handling.